distribution/src/bin/opensearch

```diff
@@ -36,14 +36,16 @@ fi
 
 # get keystore password before setting java options to avoid
 # conflicting GC configurations for the keystore tools
-unset KEYSTORE_PASSWORD
-KEYSTORE_PASSWORD=
 if [[ $CHECK_KEYSTORE = true ]] \
     && bin/opensearch-keystore has-passwd --silent
 then
-  if ! read -s -r -p "OpenSearch keystore password: " KEYSTORE_PASSWORD ; then
-    echo "Failed to read keystore password on console" 1>&2
-    exit 1
+  if [[ ! -z "${KEYSTORE_PASSWORD}" ]]; then
```

Comment:
@cwperks I think we need to update `opensearch.bat` as well (Windows)

Reply:
I tested this on windows and the changes work as expected.

To test on windows I opened a Powershell session as administrator and set the env variable KEYSTORE_PASSWORD

```
setx KEYSTORE_PASSWORD "<password>" /M
```

When the env is set the program starts up and can decrypt the keystore with this env variable. The startup script does not unset/reset this variable. 

If this env variable is not defined then the startup script will prompt the user for the keystore's password.